SpaceCore/SpaceCore.cs

```diff
@@ -94,6 +97,25 @@ private void OnUpdate(object sender, UpdateTickedEventArgs e)
             }
         }
 
+        private void OnOneSecondUpdateTicked(object sender, OneSecondUpdateTickedEventArgs e)
+        {
+            // Question: Is there an internal counter somewhere, accessible outside event handlers?
+            // If there is one, and it's accessible from TileSheetExtensions, I'd rather use that than adding another counter.
+            // But I haven't found one, so I'm using an internal counter here.
+            SecondsSinceStart++;
+            if (this.Config.DisposeOldTextures)
+            {
+                // TryDequeue not available on net452
+                if (TextureDisposalQueue.Count != 0)
+                {
+                    while (TextureDisposalQueue.Peek().Value < SecondsSinceStart)
+                    {
+                        TextureDisposalQueue.Dequeue().Key.Dispose();
+                    }
+                }
+            }
+        }
+
```

Comment:
The strategy is to check, every second, if the head of the queue has reached deadline.

This assumes that the queue is intrinsically sorted by deadline, of course. If an object with earlier deadline gets queued behind an object with a later deadline, then tough; that object will have to wait for disposal until the head object reached deadline.

This cannot happen with the current code (only one func adds to the queue, and the adding is intrinsically sorted by deadline), but might need revisiting if we want to expand the functionality of the auto-disposal facility.

Reply:
I suggest...

1. change `TextureDisposalQueue` to track `Texture2D` instead of `IDisposable`, so we can check the `IsDisposed` field;
2. queue the current `Game1.ticks` instead:
   ```cs
   SpaceCore.TextureDisposalQueue.Enqueue(new(oldTexture, Game1.ticks));
   ```
3. then dispose any textures that were queued more than 60 ticks ago:
   ```cs
   if (this.Config.DisposeOldTextures)
   {
       const cacheTime = 60; // sixty ticks per second
       while (this.TextureDisposalQueue.Count != 0)
       {
           Texture2D asset = this.TextureDisposalQueue.Peek();
           if (asset.Value > Game1.ticks + cacheTime)
               break;

           this.TextureDisposalQueue.Dequeue();
           if (!asset.IsDisposed)
               asset.Dispose();
       }
   }
   ```

That avoids the manual `SecondsSinceStart` timer, avoids making any assumptions about the deadline (since it's now controlled in one place), fixes an error if something else disposes the texture, and fixes an error when calling `Peek()` after disposing the last value.